src/update-0.8.md

```diff
@@ -0,0 +1,100 @@
+# Updating to 0.8
+
+## Dependencies
+
+Rand crates now require `rustc` version 1.36.0 or later.
+This allowed us to remove some unsafe code and simplify the internal `cfg` logic.
+
+## Core features
+
+[`Rng::gen_range`] now takes a `Range` instead of two numbers.
+This requires replacing `gen_range(a, b)` with `gen_range(a..b)` in code written
+for `rand 0.7`. If `a` or `b` were a reference, explicit dereferencing may now be
+required. Additionally, inclusive ranges are now supported: `gen_range(a, b +
+1)` can be replaced with `gen_range(a..=b)`.
+
+The `AsByteSliceMut` trait was replaced with the [`Fill`] trait. This should
+only affect code implementing `AsByteSliceMut` on user-defined types, so they
+are supported by [`Rng::fill`] and [`Rng::try_fill`]. Now, the [`Fill`] trait
+has to be implemented instead: Rather than providing a mutable byte slice, the
+user-defined type must be filled with random data.
+
+The entire [`rand::rngs::adapter`] module is now restricted to the `std` feature.
+While this is technically a breaking change, it should only affect `no_std` code
+using [`ReseedingRng`], which is unlikely to exist in the wild.
+
+## PRNGs
+
+These have seen only small changes, but noteworthy is:
+
+-   [`StdRng`] and [`ThreadRng`] now use the ChaCha12 instead of the ChaCha20
+    algorithm. This improves performance and is a value-breaking change for
+    [`StdRng`].
+-   [`StdRng`], [`SmallRng`], and [`StepRng`] now implement `PartialEq` and `Eq`.
+
+## Distributions
+
+The most widely used distributions ([`Standard`] and [`Uniform`]), were not
+significantly changed. Only the following distributions suffered breaking
+changes:
+
+-   The [`Alphanumeric`] distribution now samples bytes instead of chars. This
+    more closely reflects the internally used type, but old code likely has to
+    be adapted to perform the conversion from `u8` to `char`.
+-   The alternative implementation of [`WeightedIndex`] employing the alias method
+    was moved from `rand` to [`rand_distr::WeightedAliasIndex`]. Old code has to
+    be adapted accordingly.
```

Comment:
This sentence seems to be just padding (next item too). Maybe more useful to give a quick justification: the alias-method impl has slow enough initialisation that it's less generally useful.

> Old code has to be adapted accordingly.

Reply:
I just wanted to emphasize that this is a breaking change, but I guess that's obvious enough. I'll add your justification.